### PR TITLE
In 145765803 implement datetime picker

### PR DESCRIPTION
**Pull request by jgedeon-osi**

### CODE REVIEW

File: app/assets/javascripts/components/people/PersonEditPage.jsx [hunk 1]
```diff
@@ -110,11 +110,31 @@ export class PersonEditPage extends React.Component {
...
+
+              Try Setting The Value™!
+
+              Unlike most input fields, this one initializes with a null value and
+              it's parent component doesn't propagate prop updates down to it when
+              the API response comes back. Homeowners around the country are
+              discovering this one weird bug that makes the edit form receive that
+              "change to null" action.
+
+              Call 1-800-IF-THIS-IS-A-PROBLEM-JUST-SET-THIS-FIELD-BECAUSE-THIS-PAGE-IS-GOING-AWAY-ANYWAY.
```
Comment (AmanShurafa) [inline app/assets/javascripts/components/people/PersonEditPage.jsx]: 1- A quick research proves that `Try Setting The Value™` isn't a registered trademark like advertised.
2- Phone number doesn't seem to be working.

I strongly advise you to pull the ad until you fix the problems within it?

File: spec/features/screening/participant/editing_participant_spec.rb [hunk 2]
```diff
@@ -92,6 +92,10 @@
         expect(page).to have_field('Phone Number Type', with: marge.phone_numbers.first.type)
         expect(page).to have_field('Gender', with: marge.gender)
         has_react_select_field('Language(s)', with: marge.languages)
+        # Date of birth should not have datepicker, but limiting by field ID will break when
+        # DOB fields are correctly namespaced by participant ID. Feel free to make this more
+        # specific once that's done.
+        expect(page).not_to have_selector('.rw-select')
```
Comment (AmanShurafa) [inline spec/features/screening/participant/editing_participant_spec.rb]: I realize that this is out of the scope of this story. However, the issue need to be reported as a bug and the need to fix this test should be added as a task to the that story.

File: spec/javascripts/components/common/DateFieldSpec.jsx [hunk 3]
```diff
@@ -116,6 +116,15 @@ describe('DateField', () => {
...
 
+    it('passes the min and max props down to the DateTimePicker', () => {
+      const min = new Date('2007-06-15')
+      const max = new Date('2008-08-23')
+      component = mount(<DateField {...props} min={min} max={max} />)
+      const inputElement = component.find('DateTimePicker')
+      expect(inputElement.props().max).toBe(max)
+      expect(inputElement.props().min).toBe(min)
+    })
+
```
Comment (AmanShurafa) [inline spec/javascripts/components/common/DateFieldSpec.jsx]: Can you add a placeholder comment where this can be feature tested? e.g: Screening Information Start & End Dates. Because there are no tests whatsoever (manually or otherwise) to verify that the code in this commit achieves its purpose.

File: spec/javascripts/components/people/PersonEditPageSpec.jsx [hunk 4]
```diff
@@ -39,7 +39,8 @@ describe('PersonEditPage', () => {
       expect(component.find('InputField[label="First Name"]').props().value).toEqual('Kevin')
       expect(component.find('InputField[label="Middle Name"]').props().value).toEqual('Culkin')
       expect(component.find('InputField[label="Last Name"]').props().value).toEqual('McCallister')
-      expect(component.find('DateField[label="Date of birth"]').props().value).toEqual('11/16/1990')
+      // DateField updates broke person page which is deprecated so we are not fixing this.
```
Comment (AmanShurafa) [inline spec/javascripts/components/people/PersonEditPageSpec.jsx]: I would reword `so we are not fixing this.` to `and there is no value in fixing it.`

File: spec/javascripts/components/people/PersonShowPageSpec.jsx [hunk 5]
```diff
@@ -83,8 +83,9 @@ describe('PersonShowPage', () => {
         .toContain('517-566-2111')
       expect(component.find('ShowField[label="Phone Number Type"]').last().html())
         .toContain('Home')
-      expect(component.find('ShowField[label="Date of birth"]').html())
-        .toContain('11/16/1990')
+      // DateField updates broke person page which is deprecated so we are not fixing this.
```
Comment (AmanShurafa) [inline spec/javascripts/components/people/PersonShowPageSpec.jsx]: See above comment

File: spec/javascripts/components/common/DateFieldSpec.jsx (hunk 3 above)
Comment (AmanShurafa) [inline spec/javascripts/components/common/DateFieldSpec.jsx]: Actually, I'm adding this myself since I'm going to be the first to work with min & max.

File: spec/javascripts/components/people/PersonEditPageSpec.jsx (hunk 4 above)
Comment (aaronOSI) [inline spec/javascripts/components/people/PersonEditPageSpec.jsx]: Done

File: spec/javascripts/components/people/PersonShowPageSpec.jsx (hunk 5 above)
Comment (aaronOSI) [inline spec/javascripts/components/people/PersonShowPageSpec.jsx]: Done
